rocketpy/rocket/aero_surface/generic_surface.py

```diff
@@ -85,12 +85,80 @@ def __init__(
-        self.cD = self._process_input(cD, "cD")
-        self.cQ = self._process_input(cQ, "cQ")
-        self.cm = self._process_input(cm, "cm")
-        self.cn = self._process_input(cn, "cn")
-        self.cl = self._process_input(cl, "cl")
+        default_coefficients = self._get_default_coefficients()
+        self._check_coefficients(coefficients, default_coefficients)
+        coefficients = self._complete_coefficients(coefficients, default_coefficients)
+        for coeff, coeff_value in coefficients.items():
+            value = self._process_input(coeff_value, coeff)
+            setattr(self, coeff, value)
+
+    def _get_default_coefficients(self):
+        """Returns default coefficients
+
+        Returns
+        -------
+        default_coefficients: dict
+            Dictionary whose keys are the coefficients names and keys
+            are the default values.
+        """
+        default_coefficients = {
+            "cL": 0,
+            "cQ": 0,
+            "cD": 0,
+            "cm": 0,
+            "cn": 0,
+            "cl": 0,
+        }
+        return default_coefficients
+
+    def _complete_coefficients(self, input_coefficients, default_coefficients):
+        """Creates a copy of the input coefficients dict and fill it with missing
+        keys with default values
+
+        Parameters
+        ----------
+        input_coefficients : str, dict
+            Coefficients dictionary passed by the user. If the user only specifies some
+            of the coefficients, the remaining are completed with class default
+            values
+        default_coefficients : dict
+            Default coefficients of the class
+
+        Returns
+        -------
+        coefficients : dict
+            Coefficients dictionary used to setup coefficient attributes
+        """
+        coefficients = copy.deepcopy(input_coefficients)
+        for coeff, value in default_coefficients.items():
+            if coeff not in coefficients.keys():
+                coefficients[coeff] = value
+
+        return coefficients
+
+    def _check_coefficients(self, input_coefficients, default_coefficients):
+        """Check if input coefficients have only valid keys
+
+        Parameters
+        ----------
+        input_coefficients : str, dict
+            Coefficients dictionary passed by the user. If the user only specifies some
+            of the coefficients, the remaining are completed with class default
+            values
+        default_coefficients : dict
+            Default coefficients of the class
+
+        Raises
+        ------
+        ValueError
+            Raises a value error if the input coefficient has an invalid key
+        """
+        for key in input_coefficients.keys():
+            if key not in default_coefficients.keys():
+                raise ValueError(
+                    "Invalid coefficient name used in key! Check the "
+                    + "documentation for valid names."
+                )
```

Comment:
I won`t make it a suggestion, but just for you to think about, I think this could make the code a bit faster:

```python
invalid_keys = set(input_coefficients) - set(default_coefficients)
if invalid_keys:
    raise ValueError(
        f"Invalid coefficient name(s) used in key(s): {', '.join(invalid_keys)}. "
        "Check the documentation for valid names."
    )
```

Reply:
Good suggestion, especially because it is more informative on which keys are invalid!